#Author: Erik de Castro Lopo <erikd@mega-nerd.com>
#Description :
# Before this patch alex we generating code that produced warnings when
# compiled with GHC. This patch was submitted to the alex bug tracker:
#
#    http://trac.haskell.org/haskell-platform/ticket/141
#
# and the patch has been applied to the alex darcs repo, but there has been
# no new release yet.

Index: alex-2.3.3/templates/GenericTemplate.hs
===================================================================
--- alex-2.3.3.orig/templates/GenericTemplate.hs
+++ alex-2.3.3/templates/GenericTemplate.hs
@@ -57,10 +57,10 @@
 ALEX_IF_BIGENDIAN
   narrow16Int# i
   where
-	i    = word2Int# ((high `uncheckedShiftL#` 8#) `or#` low)
-	high = int2Word# (ord# (indexCharOffAddr# arr (off' +# 1#)))
-	low  = int2Word# (ord# (indexCharOffAddr# arr off'))
-	off' = off *# 2#
+	!i    = word2Int# ((high `uncheckedShiftL#` 8#) `or#` low)
+	!high = int2Word# (ord# (indexCharOffAddr# arr (off' +# 1#)))
+	!low  = int2Word# (ord# (indexCharOffAddr# arr off'))
+	!off' = off *# 2#
 ALEX_ELSE
   indexInt16OffAddr# arr off
 ALEX_ENDIF
@@ -74,14 +74,14 @@
 ALEX_IF_BIGENDIAN
   narrow32Int# i
   where
-   i    = word2Int# ((b3 `uncheckedShiftL#` 24#) `or#`
+   !i    = word2Int# ((b3 `uncheckedShiftL#` 24#) `or#`
		     (b2 `uncheckedShiftL#` 16#) `or#`
		     (b1 `uncheckedShiftL#` 8#) `or#` b0)
-   b3   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 3#)))
-   b2   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 2#)))
-   b1   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 1#)))
-   b0   = int2Word# (ord# (indexCharOffAddr# arr off'))
-   off' = off *# 4#
+   !b3   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 3#)))
+   !b2   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 2#)))
+   !b1   = int2Word# (ord# (indexCharOffAddr# arr (off' +# 1#)))
+   !b0   = int2Word# (ord# (indexCharOffAddr# arr off'))
+   !off' = off *# 4#
 ALEX_ELSE
   indexInt32OffAddr# arr off
 ALEX_ENDIF
